# Práctica 1.1 - Probabilidad y Estadística
# Análisis de evolución poblacional mundial
# Actividad 1 — Identificación de la fuente 

La fuente original es **United Nations, World Population Prospects (2024)**. 

**Cita:** 

> UN, World Population Prospects (2024) – processed by Our World in Data. “Annual change in population – UN WPP” [dataset]. United Nations, “World Population Prospects”; United Nations, “World Population Prospects - Interim Update” [original data]. 

**Preguntas** 

Responde brevemente: 

1. ¿Quién produjo originalmente los datos? United Nations

2. ¿Qué organización los procesa para su utilización en Our World in Data? Out World in Data

3. ¿Qué representa la variable que analizarás? annual population growth

# Actividad 2 — Carga y exploración de datos 

In [2]:
import pandas as pd
import numpy as np

In [3]:
#Leer el documento
df = pd.read_csv(r"C:\Users\nesso\Downloads\5to Semestre\Big Data\U1_1_probabilidad_estadisctica\data\raw\annual-population-growth\annual-population-growth.csv")

In [4]:
#primeras filas
df.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


In [5]:
#Dimensiones
df.shape

(38400, 5)

In [6]:
#nombres de columnas
df.columns.tolist() 

['Entity',
 'Code',
 'Year',
 'Annual change in population',
 'Annual population change (Projected)']

In [7]:
#tipos de datos
df.dtypes 

Entity                                      str
Code                                        str
Year                                      int64
Annual change in population             float64
Annual population change (Projected)    float64
dtype: object

In [8]:
#estadísticas descriptivas
df.describe() 

,Year,Annual change in population,Annual population change (Projected)
count,38400.000000,1.868800e+04,1.971200e+04
mean,2025.500000,2.063019e+06,8.375481e+05
std,43.300872,9.440859e+06,5.816745e+06
min,1951.000000,-3.315935e+06,-2.291035e+07
25%,1988.000000,1.781000e+03,-9.742750e+03
50%,2025.500000,4.689800e+04,9.350000e+01
75%,2063.000000,3.021128e+05,1.411160e+05
max,2100.000000,9.337137e+07,7.206736e+07


In [9]:
#cantidad de valores faltantes
df.isnull().sum() 

Entity                                      0
Code                                     1200
Year                                        0
Annual change in population             19712
Annual population change (Projected)    18688
dtype: int64

In [10]:
#cantidad de entidades
df["Entity"].nunique()  

256

# Preguntas 

Responde brevemente: 

1. ¿Cuántas filas y columnas tiene el dataset? 38400 rows × 5 columns

2. ¿Qué tipo de datos contiene la variable Year? int64

3. ¿Existen valores faltantes? Sí, Code: 1200, Annual change in population: 19712 y Annual population change (Projected): 18688

4. ¿Cuántas entidades diferentes contiene el dataset? 256

# Actividad 3 — Estadística descriptiva 

In [11]:
# Selección de un año histórico
año_seleccionado = 2020

In [12]:
# Filtrar datos del año y eliminar valores faltantes
df_año = df[df["Year"] == año_seleccionado].dropna(
    subset=["Annual change in population"]
)

In [13]:

variable = df_año["Annual change in population"]

In [14]:
# Estadísticos descriptivos
media = variable.mean()
mediana = variable.median()
varianza = variable.var()
desviacion_estandar = variable.std()
minimo = variable.min()
maximo = variable.max()

In [15]:
print(f"Estadísticos para el año {año_seleccionado}")
print("Media:", media)
print("Mediana:", mediana)
print("Varianza:", varianza)
print("Desviación estándar:", desviacion_estandar)
print("Mínimo:", minimo)
print("Máximo:", maximo)

Estadísticos para el año 2020
Media: 2124908.84765625
Mediana: 34293.5
Varianza: 90687335031103.64
Desviación estándar: 9522989.815761836
Mínimo: -494016.0
Máximo: 75707590.0


In [16]:
# Entidades con el valor mínimo y máximo
entidad_minima = df_año.loc[
    variable.idxmin(), ["Entity", "Annual change in population"]
]

entidad_maxima = df_año.loc[
    variable.idxmax(), ["Entity", "Annual change in population"]
]

print("Entidad con valor mínimo:")
print(entidad_minima)

print("\nEntidad con valor máximo:")
print(entidad_maxima)

Entidad con valor mínimo:
Entity                         Venezuela
Annual change in population    -494016.0
Name: 37269, dtype: object

Entidad con valor máximo:
Entity                              World
Annual change in population    75707590.0
Name: 37869, dtype: object


In [17]:
#Sacar estadisticos de:
# -estadisticas Pais 1er letra del apellido(Z) 1983-2021
# -df minimo-máximo

paises_con_z = df[df["Entity"].str.startswith("Z", na=False)]["Entity"].unique()
print("Países que inician con Z:")
print(paises_con_z)


Países que inician con Z:
<StringArray>
['Zambia', 'Zimbabwe']
Length: 2, dtype: str


In [18]:
#elegimos el pais
pais_z = "Zambia" 

In [19]:
# Filtramos por país y por el rango de años
df_z = df[(df["Entity"] == pais_z) & (df["Year"] >= 1983) & (df["Year"] <= 2021)]

In [20]:
# Sacamos las estadísticas de la variable que nos interesa
estadisticas_z = df_z["Annual change in population"].describe()
print(f"--- Estadísticas para {pais_z} (1983-2021) ---")
print(estadisticas_z)

--- Estadísticas para Zambia (1983-2021) ---
count        39.000000
mean     344230.051282
std      135632.355268
min      195024.000000
25%      203608.500000
50%      322769.000000
75%      475516.000000
max      545555.000000
Name: Annual change in population, dtype: float64


In [21]:
variable_total = df["Annual change in population"].dropna()

fila_minima = df.loc[variable_total.idxmin(), ["Entity", "Year", "Annual change in population"]]
fila_maxima = df.loc[variable_total.idxmax(), ["Entity", "Year", "Annual change in population"]]



In [22]:
print("Entidad con valor mínimo:")
print(fila_minima)

print("\nEntidad con valor máximo:")
print(fila_maxima)

Entidad con valor mínimo:
Entity                           Ukraine
Year                                2023
Annual change in population   -3315935.0
Name: 35772, dtype: object

Entidad con valor máximo:
Entity                              World
Year                                 1990
Annual change in population    93371370.0
Name: 37839, dtype: object


In [23]:
import sys

print("Python:", sys.version)
print("Pandas:", pd.__version__)
print("NumPy:", np.__version__)


Python: 3.14.0 (tags/v3.14.0:ebf955d, Oct  7 2025, 10:15:03) [MSC v.1944 64 bit (AMD64)]
Pandas: 3.0.5
NumPy: 2.5.2


In [24]:
df.loc[:,"Entity"]

0        Afghanistan
1        Afghanistan
2        Afghanistan
3        Afghanistan
4        Afghanistan
            ...     
38395       Zimbabwe
38396       Zimbabwe
38397       Zimbabwe
38398       Zimbabwe
38399       Zimbabwe
Name: Entity, Length: 38400, dtype: str

In [25]:
df.iloc[:,0]

0        Afghanistan
1        Afghanistan
2        Afghanistan
3        Afghanistan
4        Afghanistan
            ...     
38395       Zimbabwe
38396       Zimbabwe
38397       Zimbabwe
38398       Zimbabwe
38399       Zimbabwe
Name: Entity, Length: 38400, dtype: str

In [29]:
df.loc[10000]

Entity                                  Equatorial Guinea
Code                                                  GNQ
Year                                                 2051
Annual change in population                           NaN
Annual population change (Projected)              46407.0
Name: 10000, dtype: object

In [31]:
df.loc[:,["Entity","Year","Annual change in population"]]

,Entity,Year,Annual change in population
0,Afghanistan,1951,103163.0
1,Afghanistan,1952,108441.0
2,Afghanistan,1953,108919.0
3,Afghanistan,1954,111251.0
4,Afghanistan,1955,119027.0
...,...,...,...
38395,Zimbabwe,2096,NaN
38396,Zimbabwe,2097,NaN
38397,Zimbabwe,2098,NaN
38398,Zimbabwe,2099,NaN


In [36]:
df.iloc[:,[0,2,3]]

,Entity,Year,Annual change in population
0,Afghanistan,1951,103163.0
1,Afghanistan,1952,108441.0
2,Afghanistan,1953,108919.0
3,Afghanistan,1954,111251.0
4,Afghanistan,1955,119027.0
...,...,...,...
38395,Zimbabwe,2096,NaN
38396,Zimbabwe,2097,NaN
38397,Zimbabwe,2098,NaN
38398,Zimbabwe,2099,NaN


In [39]:
df.loc[10000,["Entity","Year","Annual change in population"]]

Entity                         Equatorial Guinea
Year                                        2051
Annual change in population                  NaN
Name: 10000, dtype: object

In [40]:
df.iloc[10000,[0,2,3]]

Entity                         Equatorial Guinea
Year                                        2051
Annual change in population                  NaN
Name: 10000, dtype: object

In [44]:
df.loc[año_seleccionado:,"Entity"]

2020     Asia (UN)
2021     Asia (UN)
2022     Asia (UN)
2023     Asia (UN)
2024     Asia (UN)
           ...    
38395     Zimbabwe
38396     Zimbabwe
38397     Zimbabwe
38398     Zimbabwe
38399     Zimbabwe
Name: Entity, Length: 36380, dtype: str

In [47]:
pais_m = "Mexico"
df_m = df[(df["Entity"] == pais_m) & (df["Year"] == 2020)]

In [48]:
print(df_m)

       Entity Code  Year  Annual change in population  \
22269  Mexico  MEX  2020                    1036075.0   

       Annual population change (Projected)  
22269                                   NaN  


In [50]:
df.iloc[0:5]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


In [51]:
df.iloc[:,[0,2,3]]

,Entity,Year,Annual change in population
0,Afghanistan,1951,103163.0
1,Afghanistan,1952,108441.0
2,Afghanistan,1953,108919.0
3,Afghanistan,1954,111251.0
4,Afghanistan,1955,119027.0
...,...,...,...
38395,Zimbabwe,2096,NaN
38396,Zimbabwe,2097,NaN
38397,Zimbabwe,2098,NaN
38398,Zimbabwe,2099,NaN


In [52]:
df.iloc[0:4,0:3]

,Entity,Code,Year
0,Afghanistan,AFG,1951
1,Afghanistan,AFG,1952
2,Afghanistan,AFG,1953
3,Afghanistan,AFG,1954
